File: CPP Unit 05/Lec 34/Q2.cpp
Single File Programming Question
Problem Statement



Akshitha is tasked with developing a program to determine the day of the week based on a given integer input representing a day number (1 to 7). In this scenario, each integer from 1 to 7 corresponds to a specific day of the week, as follows:



1: Sunday

2: Monday

3: Tuesday

4: Wednesday

5: Thursday

6: Friday

7: Saturday



The program should take an integer input from the user representing the day number and output the corresponding day of the week. If the input is not within the range of 1 to 7, the program should print "Invalid."



You need to create a class named Main that handles this functionality. The constructor of the class Main takes an integer argument representing the day number. The class should then determine the day of the week based on the input and print the result to the console.



You should also ensure that the class Main has a virtual destructor to allow for proper cleanup when deleting objects through a base-class pointer.

Input format :
The input is a single integer d, representing the day number.

Output format :
The output displays a string representing the corresponding day of the week. If the input d is within the range of 1 to 7, the program should output the day of the week.

If the input d is outside this range, the program should output "Invalid".



Refer to the sample output for the formatting specification.

Code constraints :
1 ≤ d ≤ 7

Sample test cases :
Input 1 :
7
Output 1 :
Saturday
Input 2 :
0
Output 2 :
Invalid
Input 3 :
9
Output 3 :
Invalid

Whitelist
Set 1:
virtual





#include <iostream>
using namespace std;

class Main {
public:
    Main(int d) {
        if (d == 1) cout << "Sunday";
        else if (d == 2) cout << "Monday";
        else if (d == 3) cout << "Tuesday";
        else if (d == 4) cout << "Wednesday";
        else if (d == 5) cout << "Thursday";
        else if (d == 6) cout << "Friday";
        else if (d == 7) cout << "Saturday";
        else cout << "Invalid";
    }

    virtual ~Main() {}
};

int main() {
    int d;
    cin >> d;
    Main obj(d);
    return 0;
}

File: Java/Unit 04/Static/Q2.java
Single File Programming Question
Problem Statement



Alex, a project manager, needs to ensure that all deadlines entered into the system are valid and determine if they fall on a weekend or a weekday. You have been tasked with writing a utility class that checks whether a given date (in dd mm yyyy format) falls on a weekend or a weekday. Alex will enter the project deadline in the format dd mm yyyy, and your program should output whether the deadline is on a Weekend or Weekday.

Input format :
The first line of input contains a single line representing the date in dd mm yyyy format.

Output format :
The first line of output prints

"<Date> is a Weekend" if the date is a Saturday or Sunday. (or)
"<Date> is a Weekday" if the date is any day from Monday to Friday.


Refer to the sample output for formatting specifications.

Code constraints :
The input date will always be a valid date.

The input will always be in the dd mm yyyy format.

2000 ≤ Year ≤ 2024

Sample test cases :
Input 1 :
06 05 2004
Output 1 :
06 05 2004 is a Weekday
Input 2 :
06 01 2024
Output 2 :
06 01 2024 is a Weekend




import java.util.Scanner;
import java.time.LocalDate;

// Utility class
class DateValidator {

    public static String getDayType(String dateInput) {

        // Split input "dd mm yyyy"
        String[] parts = dateInput.split(" ");

        int dd = Integer.parseInt(parts[0]);
        int mm = Integer.parseInt(parts[1]);
        int yyyy = Integer.parseInt(parts[2]);

        // Create LocalDate (yyyy, mm, dd)
        LocalDate date = LocalDate.of(yyyy, mm, dd);

        // Get day of week
        java.time.DayOfWeek day = date.getDayOfWeek();

        // Check if weekend
        if (day == java.time.DayOfWeek.SATURDAY ||
            day == java.time.DayOfWeek.SUNDAY) {
            return "Weekend";
        } else {
            return "Weekday";
        }
    }
}

// Main class
public class Main {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        // Input: dd mm yyyy
        String dateInput = scanner.nextLine();

        // Get result
        String result = DateValidator.getDayType(dateInput);

        // Output
        System.out.println(dateInput + " is a " + result);

        scanner.close();
    }
}
